driver/src/main/java/org/kaazing/k3po/driver/internal/control/handler/ControlServerHandler.java

```diff
@@ -229,6 +231,43 @@ public void abortReceived(final ChannelHandlerContext ctx, MessageEvent evt) thr
         robot.abort().addListener(whenAbortedOrFinished);
     }
 
+    @Override
+    public void notifyReceived(final ChannelHandlerContext ctx, MessageEvent evt) throws Exception {
+        NotifyMessage notifyMessage = (NotifyMessage) evt.getMessage();
+        final String barrier = notifyMessage.getBarrier();
+        if (logger.isDebugEnabled()) {
+            logger.debug("NOTIFY: " + barrier);
+        }
+        robot.notifyBarrier(barrier);
+        final NotifiedMessage notifiedMessaged = new NotifiedMessage();
```

Comment:
I think thats correct, the control handler notifies the barrier when it receives a notify message and then sends a notified response.  But I'll refactor so it reuses the same logic as the await so its more readable :-)  Thanks